lib/device.ex

```diff
@@ -24,8 +39,13 @@ defmodule Onvif.Device do
     field(:ntp, :string)
     field(:media_ver10_service_path, :string)
     field(:media_ver20_service_path, :string)
-    field(:services, {:array, :map}, default: [])
-    field(:auth_type, Ecto.Enum, default: :xml_auth, values: [:xml_auth, :digest_auth, :basic_auth, :no_auth])
+    embeds_many(:services, Onvif.Device.Service)
```

Comment:
do we need to do this with :scopes as well?

Reply:
Nope, it is just a list of strings.